docs/source/en/model_doc/clvp.md

```diff
@@ -0,0 +1,128 @@
+
+The abstract from the paper is the following:
+
+*In recent years, the field of image generation has been revolutionized by the appli-
+cation of autoregressive transformers and DDPMs. These approaches model the
+process of image generation as a step-wise probabilistic processes and leverage
+large amounts of compute and data to learn the image distribution.
+This methodology of improving performance need not be confined to images. This
+paper describes a way to apply advances in the image generative domain to speech
+synthesis. The result is TorToise - an expressive, multi-voice text-to-speech sys-
+tem.*
+
+Tips:
+
+1. CLVP or Contrastive Language-Voice Pretrained Transformer is an integral part of the Tortoise TTS model.
+2. CLVP can be used to compare different generated speech candidates with the provided text, and the best speech tokens are forwarded to the diffusion model.
+3. The use of the [`CLVPModel.generate()`] method is strongly recommended for tortoise usage
+
+Brief Explanation:
+
+First the [`CLVPTokenizer`] tokenizes the `text` and [`CLVPFeatureExtractor`] extracts `log-melspectrogram` from the
+desired `audio`. Then both of the text tokens and audio representations are passed to the [`CLVPConditioningEncoder`] 
+which converts both of them into `conditioning_embeds` which preserves the text information as well as the properties 
+of the voice. [`CLVPAutoRegressiveLMHeadModel`] uses those embeds to generate multiple `speech candidates`. 
+Now the speech encoder converts each speech candidate into a vector representation and the text encoder converts the 
+text tokens into a vector with similar dimension. At the end we compare each speech vector with the text vector to see
+which speech vector is most similar to the text.
+
+
+This model was contributed by [Susnato Dhar](https://huggingface.co/susnato).
+The original code can be found [here](https://github.com/neonbjb/tortoise-tts).
+
+
+Example :
+```python
+>>> import datasets
+>>> from transformers import CLVPProcessor, CLVPModel
+
+>>> # Define the Text and Load the Audio (We are taking an audio example from HuggingFace Hub using `datasets` library)
+>>> text = "This is an example text."
+
+>>> ds = datasets.load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation")
+>>> ds = ds.cast_column("audio", datasets.Audio(sampling_rate=22050))
+>>> _, audio, sr = ds.sort("id").select(range(1))[:1]["audio"][0].values()
+
+>>> # Define processor and model
+>>> processor = CLVPProcessor.from_pretrained("susnato/clvp_dev")
+>>> model = CLVPModel.from_pretrained("susnato/clvp_dev")
+
+>>> # Generate processor output and model output 
+>>> processor_output = processor(raw_speech=audio, sampling_rate=sr, text=text, return_tensors="pt")
+>>> generated_output = model.generate(input_ids=processor_output["input_ids"], input_features=processor_output["input_features"], num_beams=4, num_return_sequences=4)
```

Comment:
Out of curiosity, what does the output of this snippet look like ?

Reply:
The output looks something like this - 
```
CLVPOutput(loss=None, 
                     speech_candidates=tensor([[ 493, 6554, 3574, 2510, 8068, 2014, 1811, 1953,  612,  866, ...],
                                               [ 493, 6554, 3574, 2510, 8068, 2014, ...]]), 
                     logits_per_speech=tensor([[27.0426], [27.0426], [27.0426], [27.0426]]), 
                     logits_per_text=tensor([[27.0426, 27.0426, 27.0426, 27.0426]]), 
                     text_embeds=tensor([[ 1.4314e-02, -1.9351e-02,  2.3121e-02, -4.8773e-03,  1.5865e-02, ...]]),
                     speech_embeds=tensor([[ 0.0092, -0.0217, -0.0008,  ..., -0.0502, -0.0142,  0.0274]]), 
                     text_model_output=tensor([[ 1.1304e-01, -1.0841e+00, ...]]), 
                     speech_model_output=tensor([[ 0.9545, -0.0912,  0.2855,  ...]]))
```
<details>
<summary>The full output.</summary> 

```python
CLVPOutput(loss=None, speech_candidates=tensor([[ 493, 6554, 3574, 2510, 8068, 2014, 1811, 1953,  612,  866,  716,   75,
          289,  463,  729,  757,  334,  155, 6086, 5281, 7627, 6934, 4437, 6083,
         5548, 6374, 7899, 7282, 4334, 7925, 7414, 4264, 8143,  548, 5709, 2976,
         6915, 8120, 8042, 3788, 1462, 5151,  279,   63,   45,   45,  248],
        [ 493, 6554, 3574, 2510, 8068, 2014, 1811, 1953,  612,  866,  716,   75,
          289,  463,  729,  757,  334,  155, 6086, 5281, 7627, 6934, 4437, 6083,
         5548, 6374, 7899, 7282, 4334, 7925, 7414, 4264, 8143,  548, 5709, 2976,
         6915, 8120, 8042, 3788, 1462, 5151,  279,   63,   45,   45,  248],
        [ 493, 6554, 3574, 2510, 8068, 2014, 1811, 1953,  612,  866,  716,   75,
          289,  463,  729,  757,  334,  155, 6086, 5281, 7627, 6934, 4437, 6083,
         5548, 6374, 7899, 7282, 4334, 7925, 7414, 4264, 8143,  548, 5709, 2976,
         6915, 8120, 8042, 3788, 1462, 5151,  279,   63,   45,   45,  248],
        [ 493, 6554, 3574, 2510, 8068, 2014, 1811, 1953,  612,  866,  716,   75,
          289,  463,  729,  757,  334,  155, 6086, 5281, 7627, 6934, 4437, 6083,
         5548, 6374, 7899, 7282, 4334, 7925, 7414, 4264, 8143,  548, 5709, 2976,
         6915, 8120, 8042, 3788, 1462, 5151,  279,   63,   45,   45,  248]]), logits_per_speech=tensor([[27.0426],
        [27.0426],
        [27.0426],
        [27.0426]]), logits_per_text=tensor([[27.0426, 27.0426, 27.0426, 27.0426]]), text_embeds=tensor([[ 1.4314e-02, -1.9351e-02,  2.3121e-02, -4.8773e-03,  1.5865e-02,
         -2.3346e-02, -1.4317e-02,  2.9134e-02, -2.3624e-02,  1.2833e-02,
          2.1609e-02,  1.4279e-02, -4.7382e-02, -1.3148e-02,  5.8283e-02,
          6.5163e-02,  2.1839e-02,  3.6012e-02, -9.7155e-02,  4.5151e-02,
         -2.3969e-02,  1.6562e-02, -3.7192e-02, -3.2447e-03, -1.4288e-03,
          8.2653e-03,  5.4093e-02, -2.6695e-03, -7.2999e-03, -5.6188e-02,
         -3.8338e-03, -2.9574e-03, -3.4861e-02, -6.7110e-02, -5.7705e-03,
          4.1229e-02, -6.3675e-03, -4.9671e-02, -3.0622e-02,  3.2844e-04,
         -2.6330e-02,  4.1235e-03,  5.5138e-02,  7.0118e-02,  1.2277e-02,
          3.8854e-02, -2.1196e-02,  1.3839e-02, -2.8949e-02,  3.5235e-02,
          9.3227e-03, -4.2757e-02,  1.0806e-02,  1.5901e-02,  7.6836e-03,
          3.4587e-02,  3.2973e-02,  6.0474e-02, -2.1226e-02, -3.4089e-02,
         -5.0197e-02, -2.4359e-02, -1.2380e-02,  3.3375e-02, -8.1062e-03,
         -7.7156e-03,  7.1038e-03,  3.1392e-02,  4.6640e-02, -7.3180e-03,
          4.3779e-02, -1.6960e-02, -1.5117e-02,  2.5586e-02,  3.7352e-02,
         -3.4132e-02,  3.1433e-02,  1.9564e-02, -2.8928e-02, -1.7381e-04,
          3.0913e-02, -4.9894e-02, -3.3997e-02,  1.3342e-02,  3.1373e-02,
          1.2985e-02, -1.1399e-02, -6.5190e-03,  3.9994e-02,  2.1737e-02,
          6.7691e-03, -1.0953e-02,  3.9449e-02, -1.7635e-02,  1.6574e-02,
         -2.3856e-02,  1.1971e-02,  2.9564e-02, -2.6394e-03, -8.6834e-03,
         -1.7557e-03,  1.8431e-02,  6.5647e-04, -6.3766e-03, -4.7142e-02,
          3.9482e-03, -1.6366e-02, -2.4827e-02,  2.3237e-02, -5.9180e-02,
         -3.5822e-02,  2.1733e-02, -1.8445e-02,  2.4029e-02,  7.9115e-02,
         -3.6843e-02,  1.2961e-02,  8.1427e-03, -1.4210e-02,  1.7260e-02,
         -2.8785e-02, -3.3115e-02,  5.3764e-03, -3.2949e-02,  3.9162e-02,
          4.8495e-03, -1.8683e-02,  6.7625e-02, -3.0521e-03, -2.2333e-03,
          4.6244e-02, -1.2065e-02, -1.8493e-02, -2.9282e-02,  9.5582e-03,
         -3.2262e-03,  3.2458e-04, -4.5218e-04, -5.3894e-02,  8.6628e-04,
         -3.2914e-02, -9.1624e-02,  4.3513e-02,  2.5471e-02,  3.3250e-02,
         -4.6212e-02, -2.3290e-02,  1.0119e-02, -4.7942e-02,  1.9583e-02,
         -1.2239e-03,  3.7687e-02, -1.2075e-02,  2.0552e-02, -5.5095e-02,
         -5.5257e-03, -4.3533e-02,  2.5661e-02,  7.3329e-02,  2.6393e-02,
         -1.6354e-02,  9.6756e-03, -4.0383e-02, -9.2242e-03,  3.3228e-02,
          1.3065e-02,  4.9174e-02,  8.3968e-03, -6.2145e-02, -3.6409e-02,
          5.4959e-02, -2.0910e-02,  3.7772e-02,  3.2393e-02, -3.4478e-02,
         -3.8737e-02,  5.0290e-03, -5.8029e-02, -1.9825e-02, -3.3157e-02,
          4.9690e-02,  4.4931e-02, -8.3420e-02,  2.4922e-02,  7.8874e-03,
         -3.6144e-02,  2.4551e-02,  4.4254e-03,  2.5509e-02, -6.6171e-03,
         -3.2065e-03, -2.0712e-02,  2.2980e-02, -3.2110e-02,  4.2510e-02,
          7.3969e-03,  5.6639e-02,  1.4345e-03, -4.2451e-02, -5.0761e-02,
          4.7206e-02,  1.5252e-02, -9.2524e-02, -2.1410e-02,  2.0170e-02,
         -1.2813e-02,  4.0533e-02,  1.7524e-02, -1.6834e-02, -4.6206e-02,
          8.4806e-03, -1.1056e-01, -1.4831e-02,  1.2453e-02, -6.2650e-02,
         -1.7679e-03,  2.6613e-02, -1.9221e-03,  3.0484e-02,  7.1749e-02,
         -1.2286e-03,  5.7215e-03, -2.7739e-02,  1.3211e-02,  2.4591e-02,
         -2.2036e-02, -6.5591e-03,  1.2655e-02,  6.1935e-03, -2.6979e-02,
         -2.4898e-02,  8.4089e-03, -5.6911e-02,  1.7867e-02,  3.3950e-02,
          2.8614e-02,  3.4791e-02, -2.7430e-02,  2.1954e-02,  2.9614e-02,
         -9.5646e-03,  5.8353e-02, -1.8201e-02, -5.0677e-02, -5.8393e-02,
          8.5915e-02, -4.7181e-02, -1.0377e-01, -1.6626e-02, -1.7025e-02,
         -6.8398e-03, -1.6001e-02,  1.2928e-02,  8.5843e-03, -8.4795e-03,
         -3.9762e-03, -3.3051e-02,  3.6894e-02, -2.0566e-02,  6.7820e-02,
          1.1856e-02,  4.6988e-02,  3.6033e-02, -5.9929e-02,  2.2616e-02,
         -7.6361e-02,  8.6754e-03, -5.1745e-03, -1.2997e-02,  3.4004e-02,
         -1.4276e-02, -3.9985e-02, -4.9671e-02,  6.0027e-03, -3.0422e-02,
          2.7251e-03,  8.0034e-03,  6.4447e-02,  1.5330e-02, -8.8089e-03,
          3.6330e-02,  6.5321e-02,  3.5319e-02,  3.6987e-02, -7.1463e-02,
         -3.7034e-02,  5.5585e-02, -1.7508e-02, -7.9170e-03,  4.6540e-02,
          9.7766e-03, -1.1755e-02,  4.6567e-03,  6.4810e-02, -7.3901e-05,
          7.7875e-02, -5.0455e-02,  1.9835e-02,  2.0872e-03, -3.9767e-02,
          1.2270e-02, -9.0943e-02, -4.8599e-02,  3.9715e-02, -2.2473e-02,
         -5.6465e-02,  1.9976e-02,  1.0832e-01,  3.5667e-02, -1.7388e-02,
         -2.8761e-02, -8.4119e-03,  3.9427e-02,  1.3220e-02, -5.5776e-02,
         -4.6077e-02, -5.2729e-02, -1.0582e-02, -4.9208e-02,  1.7708e-02,
          1.0879e-02, -4.5079e-02, -3.3857e-03, -5.5967e-02, -7.5967e-03,
         -2.4035e-02, -2.9585e-02,  2.3750e-02, -3.2872e-02,  2.6260e-02,
         -6.8161e-02, -8.2091e-02,  5.5083e-02, -3.4146e-02,  3.8187e-03,
         -3.6877e-02, -8.2331e-02,  7.0028e-02,  7.6504e-03, -1.7085e-02,
         -2.3373e-02,  2.6462e-05,  1.9825e-03,  4.7848e-02,  5.4853e-02,
          3.7593e-03, -4.4413e-02, -4.3363e-02, -7.5870e-04,  6.6827e-03,
         -2.4343e-03, -4.0490e-03, -6.0792e-02, -2.0888e-02,  1.3390e-02,
         -2.9575e-02,  6.2135e-02,  4.0394e-03,  7.1728e-03, -1.0622e-02,
         -1.5952e-02,  2.4102e-02, -2.5210e-02, -3.8581e-03,  6.1953e-02,
         -9.1317e-03,  4.7870e-02, -2.8702e-02,  1.1992e-02,  5.9091e-02,
          6.8575e-02,  1.6681e-02,  3.2777e-02, -4.1140e-02, -4.0619e-02,
         -2.0637e-02,  2.0777e-02,  4.5417e-02,  1.1813e-02,  7.0266e-02,
         -7.8000e-02,  9.7893e-03, -7.5319e-02, -1.6172e-02, -9.4084e-02,
          4.0837e-02,  7.4801e-02,  1.9455e-02,  1.4869e-02, -1.0441e-02,
          4.2316e-03, -5.6432e-02, -9.1975e-02,  1.5831e-03, -1.5004e-02,
          2.7916e-02,  6.5310e-02, -4.1397e-02, -4.7972e-02, -5.0836e-02,
         -3.3395e-03,  5.3583e-03,  6.8656e-02,  2.4406e-02,  9.3877e-03,
         -5.9588e-02,  1.7763e-02, -6.8163e-02, -4.7946e-02,  2.7935e-02,
          4.5630e-02, -4.7682e-03, -3.3209e-02, -2.1726e-04,  1.1962e-02,
          5.3932e-04,  3.3163e-02, -2.4673e-02, -2.7039e-02,  2.2107e-02,
         -6.6414e-03,  6.8726e-02, -3.3301e-02, -2.0514e-02,  3.9191e-03,
          2.5291e-02,  1.0228e-01,  3.6012e-02,  1.1567e-02, -5.0426e-02,
          6.6536e-02, -4.7445e-02, -2.7731e-02, -2.1414e-02, -2.3001e-02,
         -6.4423e-03,  3.2679e-02, -6.9543e-03,  1.1892e-02,  8.0732e-02,
          5.5511e-03,  2.7851e-02, -7.3430e-03,  3.6130e-02,  3.4124e-02,
         -4.2822e-02, -4.6913e-02,  2.4506e-02, -3.8536e-03, -3.7075e-02,
         -2.5596e-02, -4.2549e-02, -8.7681e-03, -6.2843e-02,  6.1331e-02,
          1.2299e-02,  4.1636e-02, -2.8580e-02,  9.4541e-03, -2.8093e-02,
          1.5308e-02,  1.6982e-02,  4.4629e-02,  6.2166e-02,  8.0107e-03,
         -7.8036e-03,  1.4891e-02, -7.3440e-02, -1.6662e-02,  8.1937e-03,
         -3.9186e-02,  1.9318e-02,  4.3511e-02, -2.5553e-02,  1.4850e-02,
          3.5955e-03,  7.0994e-02, -7.2727e-02, -1.0716e-02,  1.2343e-02,
          4.1686e-02,  3.2475e-02,  1.1268e-02,  6.1223e-02,  9.6928e-04,
         -4.0434e-03, -3.2305e-02, -5.8383e-02,  1.6200e-02, -1.3205e-03,
          6.8471e-02,  1.8073e-02,  5.9431e-02,  5.2990e-02,  4.6523e-02,
         -4.4371e-02, -1.9722e-02, -4.0133e-03, -1.5643e-03, -1.2183e-02,
         -5.9908e-02,  1.1223e-02,  1.6309e-02, -5.8624e-02,  4.3028e-03,
          7.4061e-02, -1.8835e-03, -4.7040e-03, -7.5026e-03, -2.5499e-02,
          5.3248e-03, -4.0075e-02,  1.6070e-02, -7.0351e-02, -1.2509e-02,
          3.0641e-02,  1.5509e-02,  5.8211e-02, -1.5468e-02,  3.4654e-03,
         -4.3228e-04,  3.5192e-02, -2.3408e-03,  5.9262e-03, -1.3048e-02,
         -3.1604e-02,  5.4152e-02,  5.0083e-03,  4.5198e-02, -2.4876e-02,
          2.6298e-02, -5.0704e-02, -1.3280e-05, -2.9601e-03, -1.3774e-02,
          1.3827e-02, -1.1252e-02, -5.8927e-02,  3.5208e-03,  7.0021e-04,
         -4.0865e-02, -1.8630e-02,  7.2632e-02, -2.6748e-02, -5.0415e-02,
         -3.8879e-02, -3.8927e-02, -1.8183e-02, -1.1341e-02, -4.5729e-02,
         -4.8694e-02, -1.7229e-03,  1.3397e-02, -3.8533e-02,  8.1133e-05,
          7.2648e-03,  5.3262e-02, -4.8252e-02, -2.6227e-02, -3.1763e-02,
          9.4037e-03,  7.9214e-03,  1.4191e-02,  4.6872e-02, -8.2830e-03,
         -1.9444e-02, -7.5808e-02,  1.1030e-02,  6.2730e-03, -1.9512e-04,
         -5.3364e-02,  1.7343e-02, -6.5640e-02, -1.1467e-03, -2.0156e-02,
          3.4264e-02, -1.0777e-02,  5.1818e-02, -6.2485e-02,  2.7529e-02,
         -3.1046e-02, -3.5575e-02,  2.7607e-02, -6.9530e-03, -1.7242e-02,
          7.4364e-04, -2.1036e-02, -6.5603e-02,  2.1952e-02, -3.2873e-03,
         -4.8479e-02, -3.6954e-03, -3.5105e-03, -4.4156e-02,  4.8208e-02,
          9.0022e-03, -4.1155e-02, -2.2152e-02, -1.2424e-02, -6.8326e-02,
          9.1695e-03,  2.0886e-02,  2.1519e-02,  2.5263e-02,  4.7305e-02,
         -5.6974e-02,  2.5265e-02,  1.4220e-02,  3.3114e-02,  4.8620e-02,
         -4.5758e-02,  1.0343e-04, -4.9498e-02,  1.3613e-02,  2.0674e-02,
         -2.2411e-02,  3.7291e-03,  2.5621e-02,  1.9977e-02, -3.5752e-02,
          5.6828e-03,  1.6738e-02, -1.9070e-02, -5.8584e-03, -5.0810e-02,
         -2.6657e-02, -1.5822e-02, -2.4740e-02, -2.9082e-02, -1.8242e-02,
         -2.4104e-02,  1.1296e-02,  1.4991e-02, -6.0062e-02, -8.5589e-03,
         -1.4223e-02,  5.9083e-02,  2.2678e-02,  3.5731e-02, -1.4736e-02,
          3.9902e-02,  1.7935e-02,  5.6123e-02,  8.1745e-03,  1.5370e-02,
         -8.2877e-03, -1.7758e-02,  4.8579e-02,  4.7698e-02, -4.6527e-02,
          5.8688e-02, -1.1132e-02, -4.6669e-02,  5.8128e-02,  1.1525e-02,
         -5.5344e-03, -1.6427e-02,  1.7380e-02, -6.9703e-03,  3.7424e-02,
         -1.0411e-02, -1.5213e-03,  2.5883e-02, -8.1085e-02, -7.9051e-02,
          7.5130e-02,  3.2405e-02, -3.7833e-02, -2.8766e-02,  1.5651e-02,
         -3.2197e-02,  1.3046e-02, -1.4645e-02,  4.3360e-02,  2.9216e-02,
         -1.8301e-02,  4.3674e-05,  1.2412e-02,  4.4863e-03,  1.1281e-02,
          1.5194e-02,  1.7099e-02, -1.3697e-02,  3.9756e-02, -1.7685e-02,
         -4.2271e-02,  7.4616e-02, -3.8442e-02,  2.8379e-03, -5.0034e-03,
          6.2001e-02, -4.2820e-03,  3.8366e-02,  3.2046e-03,  1.1926e-02,
         -2.7807e-03,  4.8286e-02, -3.5401e-02,  2.5610e-02, -1.0996e-02,
          3.6183e-02,  6.3429e-03,  1.0750e-02,  4.0496e-02,  1.7161e-03,
          2.2848e-02, -5.3018e-02, -3.2075e-02,  2.9594e-03,  1.0181e-03,
         -4.9754e-02, -3.0800e-02,  5.4995e-02,  2.0397e-02,  9.7419e-03,
          6.7261e-02, -8.5558e-02, -1.1915e-02,  4.3370e-02, -7.7213e-04,
          6.1286e-02, -6.1302e-02,  4.0776e-02, -6.2949e-02, -1.1716e-03,
          7.0118e-02, -2.1938e-02, -1.1156e-02, -1.6771e-02,  3.2772e-02,
         -2.8697e-02, -2.5999e-02, -3.7409e-02,  4.7132e-02, -3.1789e-02,
         -1.7846e-02, -3.9859e-02, -5.8049e-02,  1.4900e-02, -5.6713e-02,
         -3.8913e-02,  7.1578e-03,  3.2419e-02, -1.7925e-02,  7.6573e-03,
          2.2840e-02, -3.7153e-02,  8.8402e-03, -7.4955e-02,  3.1041e-02,
         -1.8926e-02, -2.1366e-02, -2.0907e-02,  5.6120e-02, -1.1645e-02,
         -1.2440e-02, -5.8156e-04, -3.3041e-02,  2.2460e-03,  3.9882e-02,
         -1.5548e-02,  1.7636e-02, -2.6330e-02, -1.6333e-03,  1.1827e-02,
         -7.6382e-03, -6.1219e-02,  5.0989e-03]]), speech_embeds=tensor([[ 0.0092, -0.0217, -0.0008,  ..., -0.0502, -0.0142,  0.0274],
        [ 0.0092, -0.0217, -0.0008,  ..., -0.0502, -0.0142,  0.0274],
        [ 0.0092, -0.0217, -0.0008,  ..., -0.0502, -0.0142,  0.0274],
        [ 0.0092, -0.0217, -0.0008,  ..., -0.0502, -0.0142,  0.0274]]), text_model_output=tensor([[ 1.1304e-01, -1.0841e+00, -1.5109e-01, -2.5759e-01,  1.4264e+00,
          1.4684e-01,  2.5769e-01, -5.6348e-01,  2.9918e-01, -4.0581e-01,
         -4.3245e-01,  5.0997e-01, -6.3189e-01, -5.3930e-01,  6.7684e-01,
          1.1612e+00, -8.2396e-01, -7.6758e-01,  4.7404e-01,  3.4939e-02,
         -5.7839e-01, -5.9524e-02,  9.2916e-03, -4.5417e-01,  2.6532e-01,
         -6.3854e-01,  2.9461e-02,  6.0079e-01, -7.2163e-01, -2.7447e-01,
         -2.0178e-01,  5.8628e-01,  1.5785e-01, -5.4785e-01,  1.2162e+00,
          1.1488e+00,  4.2516e-01,  1.1140e+00, -3.2346e-01,  9.5041e-01,
          3.4743e-01,  6.6216e-01,  9.1393e-01,  6.4389e-01,  1.5139e+00,
          6.6763e-01, -5.5612e-01,  7.6086e-01, -1.6975e+00,  1.0047e+00,
         -5.2106e-01, -1.6738e-01, -6.3661e-01,  5.0449e-01,  1.6395e-01,
         -5.3330e-01, -2.9772e-01, -6.8418e-01,  5.4306e-01, -1.2916e+00,
         -5.0372e-01,  6.3314e-01,  2.6993e-01, -1.4578e+00,  1.1511e+00,
          1.3441e+00,  6.0860e-02, -1.3866e+00,  5.7951e-01, -2.4335e-01,
         -8.4398e-01,  1.0632e+00, -8.2076e-01, -7.9109e-01, -5.6036e-02,
         -4.5102e-01,  1.9135e-02, -6.3675e-01, -2.0610e-03, -1.6929e+00,
          4.0017e-01,  5.6044e-01,  5.3821e-01,  1.4933e-01, -3.8199e-01,
          9.6805e-01,  4.2808e-01,  1.9522e+00, -9.2460e-01,  5.8102e-01,
         -6.8716e-01, -5.4434e-01,  1.5097e-02,  1.5553e+00,  1.6621e-01,
          6.8629e-02,  3.4719e-01,  5.2159e-01, -5.5358e-02, -1.6812e-02,
          8.7730e-01, -1.3052e-01,  1.5031e-01,  5.3459e-01,  2.0397e-01,
         -1.4546e-01, -4.8598e-01, -1.3952e+00, -6.1145e-01, -7.0687e-02,
         -8.0754e-01, -1.0275e-03,  7.2050e-01,  3.1675e-01, -1.3478e+00,
          3.1059e-01,  8.8019e-01, -9.2394e-01, -2.0154e-01, -6.4581e-01,
         -1.0138e+00,  4.8565e-01,  8.9415e-01,  2.7352e-01, -2.7131e-01,
          8.7015e-01, -3.1731e-01,  6.5900e-01, -1.0599e+00, -9.8013e-02,
         -7.8063e-02,  1.3179e-02, -1.1264e+00,  4.4203e-01, -7.8457e-01,
          7.7890e-01,  1.2889e-01, -3.6647e-01, -2.4532e-01, -4.8146e-01,
          1.3246e+00, -8.1283e-01, -1.5700e-01, -1.9707e-01, -1.2031e+00,
         -1.8447e-01, -6.6795e-01,  8.0545e-01, -1.9084e-01, -5.7557e-01,
         -6.5680e-02,  9.8948e-01, -1.8742e-01, -2.2706e-01,  1.7832e-01,
         -1.0910e+00,  6.7336e-01,  1.8475e-01,  3.3384e-02, -4.1855e-01,
          1.0677e-01,  3.2295e-01, -5.1541e-01, -6.0674e-01,  5.2982e-01,
          5.1915e-01,  7.6643e-01, -5.4103e-02, -7.4361e-01,  9.5911e-01,
          2.6893e-02,  5.5878e-01,  6.9707e-01, -6.9873e-04, -9.6121e-02,
         -8.3845e-01,  3.6005e-01, -1.9107e-01, -4.0412e-02, -8.5437e-02,
         -2.6539e-02, -4.6173e-01,  6.5622e-01, -9.5305e-02,  4.6942e-01,
          9.5113e-01,  6.6069e-01, -1.0789e+00,  9.0656e-01,  6.8549e-01,
          2.0733e-01,  8.0198e-01, -1.6409e-01, -7.9103e-02,  1.8523e-01,
         -7.5506e-01,  9.4302e-01,  1.4638e+00,  1.6331e-01, -3.8627e-01,
         -3.7968e-01, -1.5955e+00, -5.5691e-01, -1.4039e-01,  3.0143e-01,
          6.1405e-01,  1.9114e+00,  5.7251e-01,  2.3457e-01, -9.2872e-01,
          3.4308e-02, -4.2970e-01, -5.9690e-01,  7.7018e-01, -1.0654e+00,
         -1.1007e+00, -7.9093e-01, -1.2701e-01,  1.5508e-01, -3.7066e-01,
          1.7222e-01, -9.6791e-01, -9.7708e-02,  9.0956e-01,  3.1103e-01,
          1.0867e+00, -3.3972e-01, -2.6791e-01,  2.3909e-01,  8.4195e-02,
         -6.8814e-01, -7.6959e-02,  8.7682e-01, -1.6591e+00,  1.5822e+00,
          9.7902e-01, -5.8671e-01, -1.1636e+00,  4.7830e-01, -1.1655e+00,
         -3.8564e-01, -7.8113e-01, -2.9123e-02, -2.4768e-02, -1.6495e+00,
          5.0148e-01,  3.0199e-01, -4.5101e-01, -4.7532e-01, -4.3112e-01,
         -8.7304e-01, -4.1905e-01,  3.3051e-02,  1.5527e+00,  4.8490e-01,
         -4.9609e-01, -4.5488e-02,  7.3066e-01, -1.3169e+00, -1.0222e-01,
         -9.4202e-01,  5.2427e-01,  1.0426e+00, -4.6209e-03,  2.4696e-01,
         -1.1779e+00,  8.4481e-01,  3.8496e-01, -1.9003e-01, -9.3425e-01,
          1.1324e+00,  3.2538e-01, -3.9695e-01,  4.5397e-01,  2.4714e-01,
          2.8244e-01,  1.1081e-01, -1.0814e-01, -5.7235e-01, -4.7093e-01,
          7.5441e-01,  1.0828e+00, -3.0639e-01, -1.3789e-01,  2.6365e-01,
          1.7629e-01, -1.0900e+00, -7.5418e-01,  8.2577e-01, -3.2322e-01,
          8.4534e-01, -4.3776e-01,  1.8623e-01,  1.3422e-01, -5.7722e-01,
         -2.8120e-01, -8.9550e-01,  7.1627e-01, -1.1609e+00, -8.5337e-01,
          2.6922e-01, -1.4096e+00,  2.2053e-01,  6.0105e-01,  1.1926e+00,
         -5.3916e-01, -1.1859e+00,  1.0021e+00, -1.4191e+00, -3.8061e-02,
          6.6510e-01, -1.1114e+00, -5.7477e-01, -7.6847e-02,  6.2780e-01,
         -4.9968e-01, -1.0228e+00, -2.5836e-01, -3.8045e-01,  1.1377e+00,
         -8.7895e-02, -6.0622e-01, -1.2090e+00, -1.3289e+00,  7.1242e-02,
          1.3798e-01, -5.9180e-01, -2.2917e-01, -3.1638e-01,  1.0604e-01,
         -3.1783e-01, -8.7450e-01,  5.8498e-01,  4.1271e-01,  2.0736e-02,
         -7.6706e-01, -4.7428e-01, -8.1569e-01,  1.4659e+00, -4.8363e-01,
         -6.8813e-01, -5.6731e-01, -4.1713e-02,  2.1084e-01, -7.6225e-01,
         -3.4383e-01, -6.3699e-01, -5.9941e-01,  7.4316e-01, -8.5111e-01,
          5.5024e-01,  7.9974e-01,  1.0435e+00,  8.9643e-01, -4.2029e-01,
         -4.2831e-01, -3.6454e-01,  1.6597e-01, -5.1127e-01,  9.6399e-01,
         -1.3285e+00,  4.2138e-01,  1.2944e+00,  7.9820e-02,  1.9959e-01,
         -1.0213e+00, -9.2749e-01, -9.5337e-01,  6.4179e-01, -3.5180e-01,
          4.6782e-01,  5.4655e-01, -1.0779e+00, -4.9122e-01, -5.9264e-01,
          6.9530e-01, -5.4231e-02, -7.4594e-01,  5.2223e-01, -2.6751e-04,
          8.9465e-02, -7.5697e-01, -1.3070e-01, -1.6128e+00, -1.1765e+00,
         -6.7704e-01,  8.3018e-03, -4.5675e-01,  6.4885e-01,  3.5668e-01,
          1.1497e-02,  5.1983e-01,  5.7147e-01, -1.2415e+00, -2.2762e-01,
         -6.0503e-01, -7.3185e-01, -1.4480e+00,  4.3204e-01, -8.7754e-02,
          6.4123e-01, -1.3501e-01,  5.2742e-01, -1.3615e+00,  9.3403e-01,
         -6.6165e-01, -2.0716e+00, -3.0755e-01, -1.4641e+00, -6.1186e-01,
          1.5808e-03, -4.7413e-01,  6.8955e-01,  3.8231e-01, -1.4452e+00,
          1.0632e+00,  6.1485e-01,  4.0806e-01,  2.8280e-01,  9.1551e-01,
         -2.5627e-01, -2.7831e-01, -1.0541e+00, -4.9212e-01,  1.0709e+00,
          6.6808e-02,  5.4647e-01, -1.3044e+00, -7.0555e-01, -1.2303e+00,
          6.5618e-02, -8.8025e-01,  3.8896e-01,  1.3263e+00,  1.6722e-01,
         -6.8813e-02, -9.5158e-01,  2.4767e-01,  4.7182e-01,  4.3702e-01,
          6.8549e-01,  3.9142e-01,  1.1603e+00,  4.4129e-01, -3.0091e-01,
          1.2299e+00,  6.0196e-01,  1.0283e+00,  3.6981e-01, -1.2827e+00,
          7.3825e-01, -5.1504e-01,  3.5037e-01, -2.4932e-01,  2.5260e-01,
         -7.0299e-01, -1.0379e+00,  1.1639e+00,  1.5443e+00,  6.3760e-01,
          5.0009e-01,  1.1622e+00,  1.2879e+00, -9.2935e-01, -6.9333e-01,
          5.1670e-01, -1.0717e+00,  5.0907e-01,  1.4440e+00, -2.0744e-01,
          8.1064e-01, -3.1794e-01, -6.0954e-01, -1.1484e+00, -1.0419e+00,
         -3.8714e-04, -4.2122e-01, -9.0942e-01, -2.2314e-01,  4.2163e-01,
          5.9931e-01,  4.3200e-01, -4.8383e-01,  6.6823e-02, -2.7679e-02,
          5.8692e-02,  2.8236e-01, -4.3621e-01,  2.0870e-01, -4.5621e-01,
          2.1001e-01, -1.5264e-01, -1.1160e+00, -1.2284e+00,  1.8489e-01,
          4.6855e-01,  8.7423e-01,  1.3521e+00,  1.5323e-01,  1.0930e+00,
          1.6479e-01,  8.9757e-01, -2.0549e-01, -3.0120e-01,  1.3157e-01,
         -1.8702e+00,  5.8276e-01, -2.9441e-02, -2.1246e-01, -2.8804e-01,
          5.9874e-01, -5.5728e-01,  4.3316e-01,  1.1613e+00, -2.1776e-01,
         -5.3231e-01, -1.6634e-01, -1.3746e-01,  1.6259e+00, -1.5163e-01,
         -5.3572e-01, -2.5915e-01, -4.8256e-02,  6.8572e-02,  1.1869e+00,
          8.4033e-01, -1.1207e+00,  6.0085e-01, -2.9172e-01,  9.6924e-01,
          2.6855e-01,  1.0431e+00,  1.2266e+00,  1.6634e+00,  7.4042e-01,
         -4.7341e-01, -4.9574e-01,  7.0231e-01,  1.7290e-01, -1.0585e+00,
         -4.3158e-01, -8.6191e-02,  1.1859e-01,  5.1765e-01,  7.1910e-01,
          1.1902e-01, -1.6298e+00,  5.1463e-01,  2.6600e-01, -7.0270e-01,
         -8.1886e-02, -6.4730e-01,  5.2601e-02,  7.1633e-01,  1.4749e-01,
          9.8028e-01,  2.5303e-01,  1.2219e-01, -8.5508e-02, -8.4791e-01,
         -1.0014e-01,  3.6142e-01, -6.4478e-01,  1.1538e-01,  9.1342e-01,
         -9.8423e-01,  1.7699e+00,  8.2066e-01, -6.5399e-01,  7.6584e-01,
         -8.5889e-02, -1.3509e+00,  1.1026e+00, -6.3409e-01,  3.2971e-01,
          3.3748e-01,  6.8624e-01,  6.0655e-02,  6.3839e-01,  1.4437e+00,
          3.4937e-01,  2.0567e-01,  2.3164e-01,  4.9758e-01,  1.0587e+00,
         -9.2649e-01, -7.0957e-01,  1.3837e+00,  3.9321e-01, -1.2852e+00,
         -6.0730e-01,  4.0311e-01, -1.6028e+00,  2.1112e-01, -2.0729e-01,
         -3.4446e-01,  2.6088e-01, -3.7121e-01,  2.3906e-01,  5.1307e-02,
         -3.7945e-01, -6.8376e-01,  7.3862e-01,  8.1869e-01, -8.7254e-02,
          5.1570e-01, -4.7565e-01,  2.7815e-01, -1.9309e+00, -1.3270e+00,
          1.4016e-01, -1.0059e+00, -8.6250e-02, -1.5892e+00, -7.8083e-02,
         -4.0082e-01,  2.6576e-01, -7.4507e-02,  1.0653e+00, -3.8540e-01,
          2.5727e-01,  1.3382e-01,  2.8701e-01, -1.7173e-01, -1.0136e+00,
         -8.7067e-03, -2.3605e-01,  3.9370e-01,  1.4391e+00, -6.0887e-01,
         -1.0342e+00, -2.5936e-01,  2.6819e-01,  2.3040e-01,  1.3113e-01,
         -8.2449e-01, -3.9876e-01,  1.5801e-01,  6.1593e-01,  6.6278e-01,
          7.8809e-02,  7.2573e-01,  4.6479e-01,  2.7323e-01,  5.0269e-01,
         -5.9338e-01, -4.1762e-01,  5.4764e-01, -3.6986e-01,  8.5034e-02,
          9.0512e-01, -1.2222e+00,  6.3164e-01,  6.6861e-01, -1.3655e-01,
         -9.7442e-01,  1.2909e+00, -1.3718e+00,  9.8099e-02,  6.9117e-01,
          2.5133e-01, -4.3623e-01,  7.6005e-01, -8.7382e-01,  2.2461e-01,
          1.1669e+00, -6.5497e-01, -3.2145e-01,  1.2436e+00,  2.8595e-01,
          1.0059e-01,  7.5831e-01,  8.9537e-01,  5.1939e-01,  8.0744e-01,
          9.7050e-01,  1.9777e-01,  7.4291e-01, -1.4462e+00, -4.7117e-01,
          9.5727e-03,  4.1212e-01, -1.1929e-01,  2.4342e-02,  1.6393e+00,
         -4.7443e-01, -8.2558e-01,  1.4521e+00,  7.4586e-02,  2.1970e-01,
          6.6264e-02,  7.5818e-01, -5.3431e-01,  1.6407e-01,  1.0581e+00,
         -6.2802e-01,  1.2587e+00, -6.1424e-01, -6.4201e-01,  7.8576e-01,
          1.6208e-01,  5.2303e-01, -8.0529e-02,  5.4583e-02,  1.3329e+00,
          5.0708e-01, -1.0640e+00, -4.7328e-01,  1.2471e+00, -8.8457e-01,
         -3.0019e-01,  4.7867e-01,  3.8796e-01, -1.4591e+00, -8.8877e-01,
         -3.4755e-01,  5.1952e-01,  1.6522e-01,  3.1727e-01, -2.9935e-01,
          3.5652e-01, -7.2055e-01, -1.2213e+00,  6.1340e-01, -6.5600e-01,
          4.0530e-01,  2.6573e-03,  4.5047e-01, -8.6987e-01, -3.1604e-01,
          5.6202e-01, -5.6348e-01, -2.0400e-01,  2.6645e-02,  8.8692e-01,
         -5.1254e-01, -4.7993e-01, -2.6951e-01, -2.7681e-01,  2.6118e+00,
          1.7540e-01, -6.2364e-01,  1.8966e-01, -8.9390e-01, -5.7723e-01,
          2.3949e-02,  1.1501e+00,  1.3376e-01,  5.6873e-01,  6.6203e-01,
         -1.3236e+00,  4.3525e-01,  4.7740e-01, -7.7098e-01, -9.9326e-01,
          4.7997e-01,  8.5743e-01, -3.2061e-01, -5.0897e-01,  3.9825e-01,
          5.2619e-01, -7.5658e-01, -4.2993e-02,  7.6593e-01,  9.3757e-01,
         -2.4531e-01,  9.0550e-02, -3.0761e-01]]), speech_model_output=tensor([[ 0.9545, -0.0912,  0.2855,  ..., -0.4973,  0.2028, -0.3339],
        [ 0.9545, -0.0912,  0.2855,  ..., -0.4973,  0.2028, -0.3339],
        [ 0.9545, -0.0912,  0.2855,  ..., -0.4973,  0.2028, -0.3339],
        [ 0.9545, -0.0912,  0.2855,  ..., -0.4973,  0.2028, -0.3339]]))
```
</details>

Reply:
Nice, thanks!
How do you explain that the speech candidates are the same, although generated via beam search ?

Reply:
> How do you explain that the speech candidates are the same, although generated via beam search ? 

(If I am not wrong here) If we set `do_sample=False` then we will get the same values even if we use beam search. Because we will take the token with the highest probability each time instead of sampling from the token distribution.

 
In the file `test_modeling_clvp.py` we have the test - `test_full_model_integration` where we have `do_sample=False`, so the output values in the test will be the same everytime.

Reply:
Hey @susnato, just a quick remark on that: even with `do_sample=False`, beam search is supposed to search over different candidates, it keeps track of the `n` most probable candidates during generation.

You can get convinced of that by using the code snippet [here](https://huggingface.co/docs/transformers/main_classes/text_generation#transformers.GenerationMixin.compute_transition_scores.example), which gives different outputs despite having `do_sample=False`.

So it should have different speech candidates! Would you mind investigating why it happens ?

BTW, don't hesitate to reach out if you need help or clarification regarding @ArthurZucker's review

Reply:
Did you manage to correct this here ?

Reply:
Hi @ylacombe, sorry I forgot to take a look at it over the weekend. Now that I am looking at it, I have a doubt - 

I ran the "Example: 2" from the [code snippet](https://huggingface.co/docs/transformers/main_classes/text_generation#transformers.GenerationMixin.compute_transition_scores.example) you provided, and no matter how many times I run - 
```python
model.generate(
    **inputs,
    max_new_tokens=5,
    num_beams=4,
    num_return_sequences=4,
    return_dict_in_generate=True,
    output_scores=True,
)
```
I am getting the same output. 
Is it not how it should be? Again I might be missing something here.

Reply:
Hi @ylacombe , there might be a misunderstanding here, I meant that I get the same (`beam search`) output every time which is different from the `greedy search` output.

The output you got is exactly same as the output that I was talking about.

Reply:
So I think Clvp works exactly as it should(just like the code snippet).

Reply:
I believe that in the snippet you provided [here](https://github.com/huggingface/transformers/pull/24745#discussion_r1324749459), you get the same `speech_candidates` using beam search with CLVP for each sample of the batch. Is this still the case here?

If not, yes, it probably works as expected

Reply:
Hi @ylacombe, I updated the generation config on ckpt and checked it again, and this is the output I got - 
```
tensor([[ 493, 6554, 3574,  ...,  716,  716,  716],
        [ 493, 6554, 3574,  ...,  716,  716,  142],
        [ 493, 6554, 3574,  ...,  716,  142,  716],
        [ 493, 6554, 3574,  ...,  716,  716,  716]])
``` 
Well, even though there is not a lot of variation here, the generated output candidates are different for a single audio.
I double checked it with the original repo and same generation_config settings and its the same.

So I think it works as expected, I think that we could get some more variation with different audio examples. 

Please let me know what you think.

Reply:
Ok in that case, it works for me! Thanks for verifying this!
 
To be extra extra careful, could you check that the first candidate is different from the last ? I've checked on the original output you provided and every candidate was the same! just want to make sure it's not the case here for the 1st and last candidate!

Reply:
Hi @ylacombe, this is the output with `max_new_tokens=10` - 
```
tensor([[ 493, 6554, 3574, 2510, 8157, 2014, 1953, 1953,  612,  493],
        [ 493, 6554, 3574, 2510, 8157, 2014, 1953, 1953, 1080,  612],
        [ 493, 6554, 3574, 2510, 8157, 2014, 1953, 1953, 1953,  612],
        [ 493, 6554, 3574, 2510, 8157, 2014, 1953, 1953,  612,  716]])
```
yes 1st and last candidate are different.

Reply:
Nice, thanks ! it's super surprising given that intuitively it should produce candidates diversified enough, super interesting finding here

Reply:
marking it as resolved since the question is answered.